velox/dwio/common/ColumnVisitors.h

```diff
@@ -312,11 +312,12 @@ class ColumnVisitor {
     return currentRow() - rowAt(rowIndex_ - 1) - 1;
   }
 
-  FOLLY_ALWAYS_INLINE vector_size_t process(T value, bool& atEnd) {
+  FOLLY_ALWAYS_INLINE vector_size_t
+  process(T value, bool& atEnd, int32_t numScanned = 0) {
```

Comment:
We should not add `numScanned` here.  Column visitor should not ever be aware of this.

Reply:
I agree with you that we need to refactor inline void `ColumnVisitor<T, TFilter, ExtractValues, isDense>::addResult` to not depend on numScanned .